src/ledger/LedgerState.cpp

```diff
@@ -747,37 +819,60 @@ LedgerState::Impl::load(LedgerState& self, LedgerKey const& key)
     }
 
     auto current = std::make_shared<LedgerEntry>(*newest);
-    mEntry[key] = current;
-
     auto impl = LedgerStateEntry::makeSharedImpl(self, *current);
+
+    // Set the key to active before constructing the LedgerStateEntry, as this
+    // can throw and the LedgerStateEntry destructor requires that mActive
+    // contains key. LedgerStateEntry constructor does not throw so this is
+    // still exception safe.
     mActive.emplace(key, toEntryImplBase(impl));
-    return LedgerStateEntry(impl);
+    LedgerStateEntry lse(impl);
+
+    // TODO(jonjove): Is std::map::operator[] exception safe?
+    // std::shared_ptr assignment is noexcept
+    mEntry[key] = current;
```

Comment:
Response to first comment is the same as above.

Reply:
Handling issues like the one mentioned in your second comment is a big part of what was added in the most recent commits. This possibility is already accounted for: see the comment on lines 824-827. Specifically, the destructor of `LedgerStateEntry` deactivates the key. Since the `LedgerStateEntry` is created before `mEntry` is updated, it is guaranteed that either both `mEntry` and `mActive` are updated or neither is updated (assuming that `operator[]` has the strong exception safety guarantee, as questioned in the `TODO`).